cee_syslog_handler/__init__.py

```diff
@@ -232,3 +232,45 @@ def format(self, log_record):
         log_record._name = self.name
         return super(NamedCeeLogger, self).format(log_record)
 
+
+class FilteredNamedCeeLogger(NamedCeeLogger):
+    """
+    This handler can be used to discard log messages that match a given regular
+    expression.
+
+    Doing the filtering on the Handler is a workaround for the legacy Python
+    `fileFormat()` configuration that does not allow filters to be specified.
+    Please only use this class as a migration helper towards using RegexFilter
+    directly.
+
+    https://docs.python.org/3/library/logging.config.html#logging-config-fileformat
+    """
+    def __init__(self, address, socket_type, name, filter_regex):
+        super(FilteredNamedCeeLogger, self).__init__(address, socket_type, name)
+        self._regex_filter = RegexFilter(filter_regex)
+
+    def filter(self, record):
+        """
+        Returns True if the record shall be logged. False otherwise.
+
+        https://github.com/python/cpython/blob/2.7/Lib/logging/__init__.py#L607
+        """
+        return self._regex_filter.filter(record) and super(FilteredNamedCeeLogger, self).filter(record)
+
+
+class RegexFilter(logging.Filter):
+    """
+    This Filter can be used to discard log messages that match a given regular expression.
```

Comment:
The way this is documented it sounds like regexes would be matched to the string using `re.match`, that is, they would only match at the beginning of the message. However, the way this is written `re.search` is used, which will match anywhere in the message. We should probably either change the documentation – e.g. to _contains a match_ – or adjust the implementation if the start-of-message behaviour is the actually desired one.

Reply:
Good catch. I have updated the docs.